packages/p2p/src/components/app.jsx

```diff
@@ -50,16 +53,38 @@ class App extends Component {
 
         /* if there is no error means its an agent else its a client */
         if (!agent_info.error) {
-            this.setState({ agent_id: agent_info.p2p_agent_info.agent_id, is_agent: true });
+            await this.setState({ agent_id: agent_info.p2p_agent_info.agent_id, is_agent: true });
         }
+        return true;
+    };
+
+    handleNotifications = updated_orders => {
+        let notification_count = 0;
+
+        updated_orders.forEach(order => {
+            const modified_order = new OrderInfo(order);
+            const is_agent_buyer = this.state.is_agent && modified_order.is_buyer;
+            const is_agent_seller = this.state.is_agent && !modified_order.is_buyer;
+            const is_client_buyer = !this.state.is_agent && modified_order.is_buyer;
+            const is_client_seller = !this.state.is_agent && !modified_order.is_buyer;
+
+            if (is_agent_buyer || is_client_seller) {
+                notification_count += modified_order.is_buyer_confirmed && 1;
```

Comment:
what is && 1 for?

Reply:
when modified_order.is_buyer_confirmed true, then notification adds 1

Reply:
the can we change the if condition `if (modified_order.is_buyer_confirmed && (is_agent_buyer || is_client_seller)) {`

Reply:
we can also make it like this
```
            if ((modified_order.is_buyer_confirmed && (is_agent_buyer || is_client_seller)) ||
                (modified_order.is_pending && (is_agent_seller || is_client_buyer))
            ) {
                notification_count++
            }
```

Reply:
seems good